src/VisualStudio/Core/Def/Implementation/TableDataSource/ProgressReporter.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.TaskStatusCenter;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
```

Comment:
This feels like a strange namespace for thsi code to be in.

Reply:
ProgressReporter is not for generic usage. designed specifically to report table source updating progress.

Reply:
I guess i don't understand why that is.  Wouldn't this just be tied to the notifications of the solution crawler?  Why is it associated with table sources?

Reply:
because.. I added the notification for error list not solution crawler. otherwise, it would have been added to solution crawler. but solution crawler is for more than errors. I also didn't made the notification for diagnostic service, then it would have added to diagnostic service not error list.